src/collector.rs

```diff
@@ -51,6 +51,19 @@ impl Collector {
         self.raw.get(command).unwrap().clone()
     }
 
+    // returns a string representation of the first capture group of the regex
+    pub fn get_regex_from_command(&mut self, command: &str, regex: &str) -> Option<String> {
+        let re = Regex::new(regex).unwrap();
+        let out = self.run_command(command);
+        let output = String::from_utf8(out.stdout).unwrap();
+        let cap = re.captures(&output).unwrap();
+        if cap.len() < 2 {
+            None
+        } else {
+            Some(String::from(cap.get(1).unwrap().as_str()).clone())
+        }
```

Comment:
Can we explore this a bit? Perhaps there's an option to let us return the whole cap object here without upsetting the borrow checker, that way we don't have to arbitrarily return only the first.